pybossa/api/__init__.py

```diff
@@ -713,3 +713,37 @@ def _get_valid_service(task_id, service_name, payload, proxy_service_config):
         'Task {} loaded for user {} failed calling {} service with payload {}'.format(task_id, current_user.id, service_name, payload))
 
     return abort(403, 'The request data failed validation')
+
+
+@jsonpify
+@login_required
+@csrf.exempt
+@blueprint.route('/task/<int:task_id>/assign', methods=['POST'])
+@ratelimit(limit=ratelimits.get('LIMIT'), per=ratelimits.get('PER'))
+def assign_task(task_id=None):
+    """Assign task to users for locked, user_pref and task_queue schedulers."""
+    projectname = request.json.get('projectname', None)
+    a_project = project_repo.get_by_shortname(projectname)
+    if not a_project:
+        return abort(400)
+
+    _, ttl = fetch_lock_for_user(a_project.id, task_id, current_user.id)
+    if not ttl:
+        return abort(403, 'A lock is required for assigning a task to a user.')
+
+    # only assign the user to the task for user_pref and task_queue scheduler
+    scheduler, _ = get_scheduler_and_timeout(a_project)
+    if scheduler in (Schedulers.user_pref, Schedulers.task_queue):
```

Comment:
This line can be formatted for 
1. avoiding indented if block 
2. logging 
ex:
```python
if not scheduler in (Schedulers.user_pref, Schedulers.task_queue):
    logger.warning("Task scheduler for project is set to %s. Cannot assign users %s for task %s", scheduler, current_user.fullname, task.id)
    return abort(403, 'Project scheduler not configured for assigning a task to a user.')
```